examples/tutorials/advanced/legends.py

```diff
@@ -0,0 +1,161 @@
+"""
+Creating legends
+================
+
+The :meth:`pygmt.Figure.legend` method creates legends, whereby auto-legends as well as
+manually created legends are supported.
+"""
+
+# %%
+import io
+
+import pygmt
+
+# %%
+# Create an auto-legend
+# ---------------------
+#
+# An auto-legend can be created for the methods :meth:`pygmt.Figure.plot`,
+# :meth:`pygmt.Figure.plot3d`, and :meth:`pygmt.Figure.histogram`. Therefore the
+# ``label`` parameter has to be specified to state the desired text for the legend entry
+# (white spaces are supported). Here, we use :meth:`pygmt.Figure.plot`, exemplary. By
+# default, the legend is placed in the Upper Right corner with an offset of 0.1
+# centimeters in both x and y directions and a box with a white fill and a 1-point
+# thick, black, solid outline is drawn around the legend. The order of the legend
+# entries (top to bottom) is determine by the plotting order. Optionally, to adjust the
+# legend, append different modifiers to the string passed to ``label``. For a list of
+# available modifiers see :gmt-docs:`gmt.html#l-full`. To create a
+# :doc:`multiple-column legend </gallery/embellishments/legend>` **+N** is used with the
+# desired number of columns.
+
+fig = pygmt.Figure()
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+
+# Plot three data points with different symbols and sizes
```

Comment:
Instead of just plotting three data points, maybe also plot one or multiple lines to show that legends also works for lines?

Reply:
Good idea! Included in commit https://github.com/GenericMappingTools/pygmt/pull/3594/commits/fc68bea0e38e40d96e0a72eed95e402a24249223.